### PR TITLE
Use OAuth2 in testing

### PR DESCRIPTION
**Pull request by AricStewart**

### CODE REVIEW

File: api/src/Tests/Cases/OAuth2Test.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        $this->setupToken = false;
+        parent::setUp();
+
+    }
+
+
+    public function testTokens(): void
+    {
+        $this->runRequest('POST', '/token', null, ['grant_type' => 'passwordx', 'username' => self::$login, 'password' => self::$password, 'client_id' => self::$client], 400);
```
Comment (mshappe) [inline api/src/Tests/Cases/OAuth2Test.php:21]: One trick: look at PHPUnit's docco into data providers to DRY this up a little. It's nifty.

**Review by mshappe — DISMISSED** [review]
See my note about using data provider methods, but other than that, this looks fine. I'm not sure what about the password part itches you, but this looks normal to me.

File: api/src/Tests/Cases/OAuth2Test.php (hunk 1 above)
Comment (AricStewart) [reply api/src/Tests/Cases/OAuth2Test.php]: Provider figured out and used

File: api/src/Tests/Cases/OAuth2Test.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    protected function setUp(): void
+    {
+        $this->setupToken = false;
+        parent::setUp();
+
+    }
+
+
+    public function provider(): array
```
Comment (mshappe) [inline api/src/Tests/Cases/OAuth2Test.php:19]: \o/

**Review by mshappe — APPROVED** [review]

**Pull request merged**